src/PKSim.Core/Snapshots/Mappers/CurveChartMapper.cs

```diff
@@ -45,9 +68,40 @@ private Task<Axis[]> snapshotAxisFrom(IReadOnlyCollection<OSPSuite.Core.Chart.Ax
          return Task.WhenAll(axis.Select(_axisMapper.MapToSnapshot));
       }
 
-      public override Task<ModelCurveChart> MapToModel(SnapshotCurveChart snapshot)
+      public override async Task<TCurveChart> MapToModel(SnapshotCurveChart snapshot, CurveChartContext curveChartContext)
+      {
+         var curveChart = new TCurveChart();
```

Comment:
need to create a chart based on type. Since we have other CurveChart throughout the program, I decided to create a generic abstract mapper here so that we can handle different types